custom_components/car_wash/binary_sensor.py

```diff
@@ -157,7 +160,23 @@ async def async_update(self):
         tmpu = self.hass.config.units.temperature_unit
         temp = wdata.attributes.get(ATTR_WEATHER_TEMPERATURE)
         cond = wdata.state
-        forecast = wdata.attributes.get(ATTR_FORECAST)
+
+        daily_response = await self.hass.services.async_call(
+            WEATHER_DOMAIN,
+            SERVICE_GET_FORECASTS,
+            {"entity_id": self._weather_entity, "type": "daily"},
+            blocking=True,
+            return_response=True,
+        )
+        if not (isinstance(daily_response, dict)):
+            self._attr_is_on = None
+            raise HomeAssistantError("daily_response is empty.")
+        if (len(daily_response) == 0) or (
```

Comment:
It seems to me, or is there an error in the logic of the conditional operator?

Reply:
oh, yeah, seems you're right!, fixed it with new commit.